charms/knative-operator/tests/unit/test_charm.py

```diff
@@ -135,53 +144,100 @@ def test_apply_resources_exception(
     )
 
 
+@pytest.mark.parametrize(
+    "container_name",
+    [
+        "knative-operator",
+        "knative-operator-webhook",
+    ],
+)
 def test_update_layer_active(
-    harness, mocked_resource_handler, mocker, mocked_metrics_endpoint_provider
+    container_name, harness, mocked_resource_handler, mocker, mocked_metrics_endpoint_provider
 ):
     harness.begin()
     # Check the initial Pebble plan is empty
-    initial_plan = harness.get_container_pebble_plan("knative-operator")
+    initial_plan = harness.get_container_pebble_plan(container_name)
     assert initial_plan.to_yaml() == "{}\n"
 
     # Check the layer gets created
-    harness.charm.on.knative_operator_pebble_ready.emit("knative-operator")
-    assert harness.get_container_pebble_plan("knative-operator")._services is not None
-
-    expected_plan = {
-        "services": {
-            "/ko-app/operator": {
-                "summary": "entrypoint of the knative-operator image",
-                "startup": "enabled",
-                "override": "replace",
-                "command": "/ko-app/operator",
-                "environment": {
-                    "POD_NAME": harness.model.app.name,
-                    "SYSTEM_NAMESPACE": harness.model.name,
-                    "METRICS_DOMAIN": "knative.dev/operator",
-                    "CONFIG_LOGGING_NAME": "config-loggig",
-                    "CONFIG_OBSERVABILITY_NAME": "config-observability",
-                },
-            }
-        },
+    harness_events = {
+        KNATIVE_OPERATOR: harness.charm.on.knative_operator_pebble_ready,
+        KNATIVE_OPERATOR_WEBHOOK: harness.charm.on.knative_operator_webhook_pebble_ready,
     }
-    updated_plan = harness.get_container_pebble_plan("knative-operator").to_dict()
+    harness_events[container_name].emit(container_name)
+    assert harness.get_container_pebble_plan(container_name)._services is not None
+
+    # TODO: This could be extracted and made cleaner
```

Comment:
Agree, we do not have to do it in this PR, but yes you may have just one layer, render it, etc.